### ISSUE TITLE
Use Intl.NumberFormat for formatting values

### ISSUE THREAD

**Issue by scurker** [scurker/currency.js#212]
Formatting is already pretty versatile, but we could also potentially leverage [`Intl.NumberFormat`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/NumberFormat) to get a lot of different formats for "free". The one downside is that it's possible that parsing and formatting are out of sync if the options aren't the same.

**Comment by kg-currenxie** [comment]
Just a heads up; `Intl` isn't available in react-native's JavaScriptCore in Android.

**Comment by mayconmesquita** [comment]
Exactly! I'm using Currency.JS to replace missing react-native's Intl API.

I got Intl working react-native changing the jscFlavor, but it increased the build size in 6MB!
